### PR TITLE
feat: onboard plugin integration service

### PR DESCRIPTION
**Pull request by utsabc**
## Onboarding Plugin Integration Service

This PR introduces onboarding of plugin integration framework to transformer, we take the approach of onboarding a new service aligning to the predefined interface to dictate the service structure

Major Piece of event order orchestration for transformer <> server contract lie in `pluginAdapter.ts` which after passing the events to integration plugin and handles the consistent structuring for responding back to outer layer

The `pluginAdapter` uses integration-store to get integration-plugin to do either processor/router transformer

Further we updated type definitions referencing from `integrations-lib` which acts as a common library

Resolves
INT-837

> Description here

## Type of change
- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Related issues

> Fix [#1]() 

### Development

- [ ] Lint rules pass locally
- [ ] The code changed/added as part of this pull request has been covered with tests
- [ ] All tests related to the changed code pass in development

### Code review 

- [ ]  This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached
- [ ] "Ready for review" label attached to the PR and reviewers mentioned in a comment
- [ ] Changes have been reviewed by at least one other engineer
- [ ] Issue from task tracker has a link to this pull request 

## Summary by CodeRabbit

- **New Features**
  - Introduced a new `PluginAdapter` for data transformation at processor and router levels.
  - Added a new `PluginIntegrationService` to handle plugin destinations.
  - Implemented a new constant `PLUGIN_DEST` for integration services.

- **Bug Fixes**
  - Fixed a syntax error in the `CHANNELS` object definition.

- **Documentation**
  - Updated inline comments for ESLint rule exceptions.

- **Refactor**
  - Enhanced the `ServiceSelector` class to support plugin destinations.
  - Modified the `getTestThreshold` function for better property access.
  - Streamlined the handling of `batchedRequest` in `DestinationPostTransformationService`.
  - Adjusted imports for `Metadata` to use the `@rudderstack/integrations-lib` package.

- **Tests**
  - Expanded test cases to include the `IsProcessorEnabled` property.
  - Updated integration tests to support plugin destinations and mock Axios adapters.

- **Chores**
  - Updated type imports and renamed `Metadata` to `TransformedOutput` in type definitions.

### CODE REVIEW

File: src/services/userTransform.ts [hunk 1]
```diff
@@ -47,7 +47,7 @@ export class UserTransformService {
     }
     const responses = await Promise.all<FixMe>(
       Object.entries(groupedEvents).map(async ([, destEvents]) => {
-        const eventsToProcess = destEvents as ProcessorTransformationRequest[];
+        const eventsToProcess = destEvents as any[];
```
[the remark this answers is not in the document]
Comment (utsabc) [reply src/services/userTransform.ts:50]: Ignore it

**Comment by koladilip** [comment]
can you split this into multiple PRs for easy review

**Comment by devops-github-rudderstack** [comment]
This PR is considered to be stale. It has been open for 20 days with no further activity thus it is going to be closed in 7 days. To avoid such a case please consider removing the stale label manually or add a comment to the PR.

**Comment by devops-github-rudderstack** [comment]
This PR is considered to be stale. It has been open for 20 days with no further activity thus it is going to be closed in 7 days. To avoid such a case please consider removing the stale label manually or add a comment to the PR.

**Comment by devops-github-rudderstack** [comment]
This PR is considered to be stale. It has been open for 20 days with no further activity thus it is going to be closed in 7 days. To avoid such a case please consider removing the stale label manually or add a comment to the PR.

**Pull request closed without merging**